### ISSUE TITLE
[Package]: "Shotcut" Package Request

### ISSUE THREAD

**Issue by hansm629** [termux/termux-packages#20021]
### Why is it worth to add this package?

Currently, various advanced graphics programs such as `Blender 3D`,` Inkscape`, `Gimp`, `RawTherapee`, etc. have been ported as Termux native packages and can be used in the XFCE4 desktop environment.

<details><summary>Screenshots</summary>
<p>

![Screenshot_2024-05-04_17-01-26](https://github.com/termux/termux-packages/assets/101167173/6f4d47cf-3ec6-48c3-9f7b-88f60e4dcb8a)
![Screenshot_2024-05-04_17-02-05](https://github.com/termux/termux-packages/assets/101167173/b6ee839f-5233-4e10-b9b1-e4d9adb69128)
![Screenshot_2024-05-04_17-03-50](https://github.com/termux/termux-packages/assets/101167173/dd75525d-0265-4b25-9dfa-1b10204b89aa)
![Screenshot_2024-05-04_17-05-33](https://github.com/termux/termux-packages/assets/101167173/67c1827e-91d2-440d-9da2-531f587db039)

</p>
</details> 

but
There is no video editing program.

For flagship smartphone SoC `Snapdragon 8 Gen3` is

Compared to `Apple M1`
Show better single-core performance within the margin of error
Multi-core performance is only 8% inferior.

`shotcut` is a video editing program that supports advanced features.
If it is a `flagship SoC, `video editing will be sufficient even with `zink` & `virgl` acceleration.
![68747470733a2f2f7777772e73686f746375742e6f72672f6173736574732f696d672f73637265656e73686f74732f53686f746375742d31382e31312e31382e706e67](https://github.com/termux/termux-packages/assets/101167173/7bdd77ef-74f6-43fb-952b-fd632e60ff45)

SoCs on smartphones and tablet PC will continue to improve, 
so I think XFCE4 desktop usability will go further here if we have a video editing program for Termux native.

Please review it.
Thank you.

### Home page URL

https://github.com/mltframework/shotcut

### Source code URL

https://github.com/mltframework/shotcut

### Packaging policy acknowledgement

- [X] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

**Comment by knyipab** [comment]
I tried to build it with TUR (https://github.com/knyipab/tur/commit/8f0d79da1ee6491a6dbe461230aeb6501940ee95). It does compile and the app can start but does not render video. This is because the current `qt5` packages do not support OpenGL and there is no easy workaround (https://github.com/termux-user-repository/tur/issues/848). 

Let's wait for Qt6 with OpenGL support. truboxl is working on the PR: #19464 . When Qt6 packages are ready, I may retry building `shotcut` (to TUR? or make PR if termux-pacakges welcomes that).

**Comment by hansm629** [comment]
@knyipab 
Thank you for answer.

Come to think of it, even in `PRoot Debian,` when trying to video rendering  in `Shotcut`,
`Error: the job XML is empty!`
It fails with the above error generated. Is this the same reason?

Even if you install all `qt6` packages in the PRoot environment and try video rendering again, the same problem persists.
Error: the job XML is empty!
The above message was generated and then failed.



Is there a possibility that `Kdenlive` will also be ported to Termux in the future?

**Comment by knyipab** [comment]
That was few months ago and I couldn't quite recall, but that doesn't seem to be the error I encountered. Besides, because we will compile from the source, there should be more room to fix errors and apply workaround. 

kdenlive is harder to package. It requires KF5 (transiting to KF6) in addition to MLT. Again, KF5/6 needs OpenGL-supported Qt in compile time. There's hope given the demand for KF6-related packages e.g. plasma desktop, libreoffice, etc. But be patient.

**Comment by hansm629** [comment]
@knyipab 
Thank you for answer.

First of all, I am more familiar with using Shotcut than with Kdenlive.
I think just porting Shotcut will be enough.

Let's wait for the good news that Shotcut will be released soon along with Qt6.

We are always grateful that the Termux computing environment continues to evolve thanks to people like you. :)

**Comment by TomJo2000** [comment]
<sup>(For tracking/reference purposes)</sup>
Depends on:
- #10854

May be possible to package after merger of:
- #19464

**Comment by hansm629** [comment]
@knyipab 
hello sir :)
Now that qt6 has been merged, can we solve the problem of video not being rendered in Shotcut?

**Comment by knyipab** [comment]
My previous patches on MLT still works with qt6 and it does build. 
However, for shotcut, there are more qt6 packages needed to compile. I think we don't have most of them. You can make package requests

According to archlinux build script, https://gitlab.archlinux.org/archlinux/packaging/packages/shotcut/-/blob/main/PKGBUILD?ref_type=heads. It needs 
```
'qt6-declarative' 'qt6-imageformats' 'qt6-multimedia' 'qt6-translations' 'qt6-charts'
```

**Comment by knyipab** [comment]
This issue should be closed by #20353.

**Comment by TomJo2000** [comment]
Wasn't directly linked so it didn't get autoclosed.
